Emby.Server.Implementations/HttpServer/WebSocketConnection.cs

```diff
@@ -244,10 +246,37 @@ public void Dispose()
         /// <param name="dispose"><c>true</c> to release both managed and unmanaged resources; <c>false</c> to release only unmanaged resources.</param>
         protected virtual void Dispose(bool dispose)
         {
+            if (_disposed)
+            {
+                return;
+            }
+
             if (dispose)
             {
                 _socket.Dispose();
             }
+
+            _disposed = true;
+        }
+
+        /// <inheritdoc />
+        public async ValueTask DisposeAsync()
+        {
+            await DisposeAsyncCore().ConfigureAwait(false);
+            Dispose(false);
```

Comment:
The socket won't get disposed?
Dispose it in `DisposeAsyncCore`

Reply:
The socket will only get disposed if it has received a close frame, no?
https://source.dot.net/#System.Net.WebSockets/System/Net/WebSockets/ManagedWebSocket.cs,353

Reply:
Went ahead and stepped through the code and found `SendCloseFrameAsync` successfully sends the close frame and changes `_receivedCloseFrame` to true, this allows `DisposeCore()` to get called and everything within that function executes and it appears to be disposing of everything correctly.

Really appreciate that link to dig into the source code, that was really enlightening and something I don't do nearly enough of...

Reply:
Would that be covered by https://github.com/jellyfin/jellyfin/blob/master/Emby.Server.Implementations/Session/SessionWebSocketListener.cs? 

When I close the web client, the server uses `SessionWebSocketListener.RemoveWebSocket` to call `webSocket.Closed -= OnWebSocketClosed;`. This sets the websocket's state to `CloseReceived` which will then cause `CloseAsync` to get called on the websocket [here](https://github.com/jellyfin/jellyfin/blob/master/Emby.Server.Implementations/HttpServer/WebSocketConnection.cs#L148). `CloseAsync` eventually sets the close frame [here](https://source.dot.net/#System.Net.WebSockets/System/Net/WebSockets/ManagedWebSocket.cs,1211). 

When I abruptly shutdown the server, in the browser console, it shows the websocket as being closed then nulls it out. It seems like there's something in the client that would deal with the websocket when a connectivity issue occurs.

Reply:
@Bond-009 does my response satisfy concerns? can we resolve this thread or is more investigation required?

Reply:
I would prefer an explicit call to the socket dispose function in DisposeAsyncCore, better safe than sorry

Reply:
Sorry for the delay on this, just did a big cross country move. Will get back to this soon.